mlflow/models/cli.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import click
+import posixpath
+
+from mlflow.models import Model
+from mlflow.models.flavor_backend_registry import get_flavor_backend
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils.file_utils import TempDir
+from mlflow.utils import cli_args
+
+_logger = logging.getLogger(__name__)
+
+
+@click.group("models")
+def commands():
+    """
+    Deploy MLflow models locally.
+
+    To deploy a model associated with a run on a tracking server, set the MLFLOW_TRACKING_URI
+    environment variable to the URL of the desired server.
+    """
+    pass
+
+
+@commands.command("serve")
+@cli_args.MODEL_URI
+@click.option("--port", "-p", default=5000, help="Server port. [default: 5000]")
+@click.option("--host", "-h", default="127.0.0.1", help="Server host. [default: 127.0.0.1]")
+@cli_args.NO_CONDA
+def serve(model_uri, port, host, no_conda=False):
+    """
+    Serve a model saved with MLflow by launching a webserver on the specified host and port. For
+    information about the input data formats accepted by the webserver, see the following
+    documentation: https://www.mlflow.org/docs/latest/models.html#model-deployment.
+    """
+    return _get_flavor_backend(model_uri, no_conda=no_conda).serve(model_uri=model_uri, port=port,
+                                                                   host=host)
+
+
+@commands.command("predict")
+@cli_args.MODEL_URI
+@click.option("--input-path", "-i", default=None,
+              help="CSV containing pandas DataFrame to predict against.")
+@click.option("--output-path", "-o", default=None,
+              help="File to output results to as json file. If not provided, output to stdout.")
+@click.option("--content-type", "-t", default="json",
+              help="Content type of the input file. Can be one of {'json', 'csv'}.")
+@click.option("--json-format", "-j", default="split",
+              help="Only applies if the content type is 'json'. Specify how the data is encoded.  "
+                   "Can be one of {'split', 'records'} mirroring the behavior of Pandas orient "
+                   "attribute. The default is 'split' which expects dict like data: "
+                   "{'index' -> [index], 'columns' -> [columns], 'data' -> [values]}, "
+                   "where index  is optional. For more information see "
+                   "https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_json"
+                   ".html")
+@cli_args.NO_CONDA
+def predict(model_uri, input_path, output_path, content_type, json_format, no_conda):
```

Comment:
I think this needs to take kwargs because the user may want to pass additional arguments to downstream predict methods.

Reply:
That's a good point. The predict on MLflow models officially does not take any extra arguments even thought the downstream method could. I think it's better to leave this as is for now and add extra arguments later if we need to.

Reply:
That works. Would we be OK with adding`...` to R's `mlflow_predict()` now? It's a simple change and it would make scoring bigger data frames feasible for Keras (`predict` defaults to a batch size of 32 which renders it unusable in many cases.)

Reply:
(This probably belongs in a separate PR actually)

Reply:
Yeah that makes sense! Both adding the ```...`` and putting it into a follow up PR :)